datanode/api/trading_data_v2.go

```diff
@@ -1099,19 +1099,39 @@ func (t *TradingDataServiceV2) ListMarkets(ctx context.Context, req *v2.ListMark
 func (t *TradingDataServiceV2) ListSuccessorMarkets(ctx context.Context, req *v2.ListSuccessorMarketsRequest) (*v2.ListSuccessorMarketsResponse, error) {
 	defer metrics.StartAPIRequestAndTimeGRPC("ListSuccessorMarkets")()
 
-	markets, err := t.marketsService.ListSuccessorMarkets(ctx, req.MarketId, req.IncludeFullHistory)
+	pagination, err := entities.CursorPaginationFromProto(req.Pagination)
+	if err != nil {
+		return nil, formatE(ErrInvalidPagination, err)
+	}
+
+	markets, pageInfo, err := t.marketsService.ListSuccessorMarkets(ctx, req.MarketId, req.IncludeFullHistory, pagination)
```

Comment:
Might be worth moving the marketID validation to be before this call and not inside it so that we can use the standard `ErrEmptyMissingMarketID` and `ErrInvalidMarketID` errors, which will also unwind into a `400` http-error and not `500`.